app/controllers/backoffice/avis_controller.rb

```diff
@@ -67,6 +68,20 @@ def update_params
     params.require(:avis).permit(:answer)
   end
 
+  def redirect_if_no_sign_up_needed
+    avis = Avis.find(params[:id])
+
+    # a gestionnaire is authenticated ... lets see if it can view the dossier
+    if current_gestionnaire.present?
```

Comment:
2 trucs :
- on a besoin du return sur des `redirect` ?
- on peut pas réécrire comme ça ?

```ruby
if current_gestionnaire.present?
  # a gestionnaire is authenticated... lets see if it can view the dossier

  redirect_to backoffice_dossier_url(avis.dossier)
else if  avis.gestionnaire.present? && avis.gestionnaire.email == params[:email]
  # the avis gestionnaire has already signed up and it sould sign in

  redirect_to new_gestionnaire_session_url
end
```